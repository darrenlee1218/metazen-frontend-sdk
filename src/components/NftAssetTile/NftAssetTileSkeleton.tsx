import React, { FC } from 'react';

import Skeleton from '@mui/material/Skeleton';
import Typography from '@mui/material/Typography';

import { boxCreator } from '@components/boxCreator';

const SkeletonRoot = boxCreator({
  p: 1,
  borderRadius: 4,
  backgroundColor: (theme) => theme.palette.action.hover,
});

const NftAssetDescriptor = boxCreator({
  mt: 1,
  display: 'flex',
});

export const NftAssetTileSkeleton: FC = () => {
  return (
    <SkeletonRoot>
      <Skeleton
        variant="rectangular"
        sx={{
          width: '100%',
          aspectRatio: '1 / 1',
          borderRadius: 4,
          height: 'unset',
        }}
      />
      <NftAssetDescriptor>
        <Typography variant="h5" sx={{ flexGrow: 1 }}>
          <Skeleton />
        </Typography>
      </NftAssetDescriptor>
    </SkeletonRoot>
  );
};
