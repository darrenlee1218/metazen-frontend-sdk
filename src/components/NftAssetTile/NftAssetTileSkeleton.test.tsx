import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { NftAssetTileSkeleton } from './NftAssetTileSkeleton';
import { TestProviders } from '@utils/testing/TestProviders';

afterEach(cleanup);

it('matches snapshot', () => {
  const { container } = render(<NftAssetTileSkeleton />, { wrapper: TestProviders });
  expect(container).toMatchSnapshot();
});
