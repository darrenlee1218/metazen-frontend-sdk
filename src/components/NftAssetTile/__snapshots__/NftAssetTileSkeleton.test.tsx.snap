// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-126tmqt"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-xsbkj6-MuiSkeleton-root"
    />
    <div
      class="MuiBox-root css-1t1ny2y"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-zpm2l5-MuiTypography-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1h2g3k5-MuiSkeleton-root"
        />
      </h5>
    </div>
  </div>
</div>
`;
